### ISSUE TITLE
Feature request: Pop up numeric input widget?

### ISSUE THREAD

**Issue by deader69** [jmoenig/Snap--Build-Your-Own-Blocks#772]
Snap! easy for kids to use... right up to the moment where they have to input numeric values.

The oval boxes are small, hard to click. And even once the kids click them, they have to go from 'mouse' mode to 'keyboard' mode.

Would be nice if the input boxes expanded when clicked (like they do when you drag/drop expressions into them).

Even BETTER would be a little popup numeric editor, which would look a lot like a calculator keypad.

This would let kids 'type' numbers by clicking on the buttons of the calculator. Besides 0-9 in a 3x3 arrangement, there could be +/- keys to toggle the sign, and maybe a 'clear' button to reset to zero.

Young kids struggle with this more than you might realize, it's a real bottleneck in an otherwise fluid UI experience. It's hard for many even to find the '-' key on the keyboard.

The only time a child should HAVE to go to the keyboard is to type in names/strings (ideally).

**Comment by jmoenig** [comment]
You can select "Input sliders" in the settings menu. If activated clicking (or tabbing into) an input slot pops up a littler slider that lets you "scrub" integer values. There are lots of issues regarding "scrubbing" (range, precision, scale etc.), this is just an experimental feature at this time.

**Comment by deader69** [comment]
I know about the sliders (for variables), but I was referring to the problems kids have just typing in numeric CONSTANTS into various blocks.

The clicking together of blocks is so easy... but then they hit a road bump trying to type numbers in.

Just trying to think of ways that they can keep the mouse in their hands, and only fumble with the keyboard when naming things.

**Comment by jmoenig** [comment]
Yep, the sliders I was pointing out are exactly addressing this issue, especially for touch devices. They let you "scrub" any input slot with the mouse, without ever having to use the keyboard.

**Comment by deader69** [comment]
But sliders only work on variables, right?

I'm talking about kids trying to type in '-100' into a 'move' command block.
Seems easy to us, but young kids fumble around quite a bit with it.

**Comment by jmoenig** [comment]
But I'm telling you: Enable input sliders!

![input sliders 1](https://cloud.githubusercontent.com/assets/2596285/7123060/c3f69c20-e21f-11e4-8fb3-1db084d1eb0c.png)

then you get sliders for *every* input slot:

![input sliders 2](https://cloud.githubusercontent.com/assets/2596285/7123071/d12b887e-e21f-11e4-9e3f-647d79745d17.png)

(you can still type, so you get either way). Cool, ey?

**Comment by deader69** [comment]
Duh-oh! NOW I understand!
